curriculum/challenges/english/03-front-end-development-libraries/redux/register-a-store-listener.md

```diff
@@ -32,19 +32,26 @@ assert(
 There should be a listener function subscribed to the store using `store.subscribe`.
 
 ```js
-(getUserInput) => assert(getUserInput('index').match(/store\s*\.\s*subscribe\(/gm));
+assert.match(code,/store\s*\.\s*subscribe\(/gm);
 ```
 
 The `store.subscribe` should receive a function.
 
 ```js
-(getUserInput) => assert(getUserInput('index').match(/(\s*function\s*)|(\s*\(\s*\)\s*=>)/gm)) 
+assert.match(code,/(\s*function\s*)|(\s*\(\s*\)\s*=>)/gm);
+assert.notMatch(code,/store\.subscribe\(.+\(\)\)/)
```

Comment:
```suggestion
```
This will always catch the function call before the next test/assert and give the "wrong" hint.